cmd/nerdctl/run_network_linux_test.go

```diff
@@ -140,6 +143,78 @@ func valuesOfMapStringString(m map[string]string) map[string]struct{} {
 	return res
 }
 
+func TestRunPortWithNoHostPort(t *testing.T) {
+	type testCase struct {
+		containerPort    string
+		runShouldSuccess bool
+	}
+	testCases := []testCase{
+		{
+			containerPort:    "80",
+			runShouldSuccess: true,
+		},
+		{
+			containerPort:    "80-81",
+			runShouldSuccess: true,
+		},
+		{
+			containerPort:    "49153-61000",
+			runShouldSuccess: false,
+		},
+		{
+			containerPort:    "80-81/tcp",
+			runShouldSuccess: true,
+		},
+	}
+	tID := testutil.Identifier(t)
+	for i, tc := range testCases {
+		i := i
+		tc := tc
+		tcName := fmt.Sprintf("%+v", tc)
+		t.Run(tcName, func(t *testing.T) {
+			testContainerName := fmt.Sprintf("%s-%d", tID, i)
+			base := testutil.NewBase(t)
+			defer base.Cmd("rm", "-f", testContainerName).Run()
+			pFlag := tc.containerPort
+			cmd := base.Cmd("run", "-d",
+				"--name", testContainerName,
+				"-p", pFlag,
+				testutil.NginxAlpineImage)
+			var result *icmd.Result
+			stdoutContent := ""
+			if tc.runShouldSuccess {
+				cmd.Base.T.Helper()
+				result = cmd.Run()
+				stdoutContent = result.Stdout() + result.Stderr()
+				assert.Assert(cmd.Base.T, result.ExitCode == 0, stdoutContent)
+			} else {
+				cmd.AssertFail()
+				return
+			}
+			logrus.Infof("The output is %s", stdoutContent)
+			regexExpression := regexp.MustCompile(".*?the auto allocate port is from (?P<portNumber>\\d*?):80.*?")
+			match := regexExpression.FindStringSubmatch(stdoutContent)
```

Comment:
Why not use `nerdctl port` to check the port number?

Reply:
Because we also need regex to get the endpoint from the `nerdctl port` output